### PR TITLE
Command in dbstructure.php to convert all tables from MyISAM to InnoDB

### PR DESCRIPTION
**Pull request by annando**
This should make the transition from MyISAM to InnoDB much more easier.

### CODE REVIEW

File: include/dbstructure.php [hunk 1]
```diff
@@ -8,12 +8,37 @@
...
+	if (!dbm::is_result($r)) {
+		echo t('There are no tables on MyISAM.')."\n";
+		return;
+	}
+
+	foreach ($r AS $table) {
+		$sql = sprintf("ALTER TABLE `%s` engine=InnoDB;", dbesc($table['TABLE_NAME']));
+		echo $sql."\n";
+
+		$result = @$db->q($sql);
```
Comment (Quix0r) [inline include/dbstructure.php]: warning suppression is maybe not what we want?

File: include/dbstructure.php [hunk 2]
```diff
@@ -1741,11 +1765,12 @@ function dbstructure_run(&$argv, &$argc) {
 	echo "dryrun		show database update schema queries without running them\n";
 	echo "update		update database schema\n";
 	echo "dumpsql		dump database schema\n";
+	echo "innodb		convert all tables from MyISAM to InnoDB\n";
```
Comment (Quix0r) [inline include/dbstructure.php]: Let's rename it to `toinnodb`?

File: mod/admin.php [hunk 3]
```diff
@@ -447,7 +447,7 @@ function admin_page_summary(App $a) {
...
 	if (dbm::is_result($r)) {
 		$showwarning = true;
-		$warningtext[] = sprintf(t('Your DB still runs with MyISAM tables. You should change the engine type to InnoDB. As Friendica will use InnoDB only features in the future, you should change this! See <a href="%s">here</a> for a guide that may be helpful converting the table engines. You may also use the <tt>convert_innodb.sql</tt> in the <tt>/util</tt> directory of your Friendica installation.<br />'), 'https://dev.mysql.com/doc/refman/5.7/en/converting-tables-to-innodb.html');
+		$warningtext[] = sprintf(t('Your DB still runs with MyISAM tables. You should change the engine type to InnoDB. As Friendica will use InnoDB only features in the future, you should change this! See <a href="%s">here</a> for a guide that may be helpful converting the table engines. You may also use the command <tt>php include/dbstructure.php innodb</tt> of your Friendica installation for an automatic conversion.<br />'), 'https://dev.mysql.com/doc/refman/5.7/en/converting-tables-to-innodb.html');
 	}
 	// MySQL >= 5.7.4 doesn't support the IGNORE keyword in ALTER TABLE statements
 	if ((version_compare($db->server_info(), '5.7.4') >= 0) AND
```
Comment (Quix0r) [inline mod/admin.php]: OT: Ah, here `version_compare()` is being used. Should be generalized (my TODO?).

File: include/dbstructure.php (hunk 1 above)
Comment (annando) [inline include/dbstructure.php]: It's just some "copy & paste" :-)

File: include/dbstructure.php (hunk 2 above)
Comment (annando) [inline include/dbstructure.php]: @Hypolite what do you think?

Comment (Hypolite) [inline include/dbstructure.php]: Agreed with @Quix0r, the current label isn't an action.

**Pull request merged**